### ISSUE TITLE
Spread syntax causes ESLint TypeError

### ISSUE THREAD

**Issue by Dev-CH** [urielvan/eslint-plugin-newline-destructuring#1]
When destructuring an object and including a spread to collate any remaining values ESLint throws `TypeError: Cannot read property 'type' of undefined`. If the destructure is already correctly formatted on new lines, ESLint runs successfully.

```
{
...
"rules" : {
  "newline-destructuring/newline": [
    "error",
      {
        "items": 1
      }
    ]
  }
...
}
```

The following example throws a `TypeError`
```js
const foo = {first: 1, second: 2};
const {first, ...bar} = foo;
```

**Comment by urielvan** [comment]
@Dev-CH Would you please provide a mini showcase to reproduce? Since in my own project, with eslint@7.21.0 and @typescript-eslint/parser,  it doesn't throw such error.

**Comment by Dev-CH** [comment]
@urielvan, I have set up a project [here](https://github.com/Dev-CH/eslint-plugin-newline-destructuring-type-error) showing the `TypeError`.

**Comment by urielvan** [comment]
@Dev-CH I've released a new version to fix this.

Btw, maybe you should consider replacing babel-eslint with @babel/eslint-parser, since the original one has been deprecated.
